python_modules/libraries/dagster-aws/dagster_aws/ecs/client.py

```diff
@@ -1,259 +1,138 @@
+            },
         )
-        task = taskdict["tasks"][0]
-        container = task["containers"][0]
-        running_task_arn = container["taskArn"]
-        self.tasks.append(running_task_arn)
-        self.tasks_done.append(False)
-        self.offset += 1
+        task_arns = [task["taskArn"] for task in response["tasks"]]
 
-    def check_if_done(self, offset=0):
-        """
-        polls ECS to see if the task is complete, pulling down logs if so
+        retries = self.max_polls
+        while retries > 0:
+            response = self.client.describe_tasks(cluster=self.cluster, tasks=task_arns)
+            tasks = [task for task in response["tasks"]]
 
-        TODO:  handle pagination for 100+ tasks
-        Args:
-             offset (int): which task to check for
+            if all(task["lastStatus"] == "STOPPED" for task in tasks):
+                break
 
-        Returns:
-             boolean corresponding to completion
-        """
-        if self.tasks_done[offset]:
-            return True
-        response = self.ecs_client.describe_tasks(tasks=self.tasks, cluster=self.cluster)
-        resp = response["tasks"][offset]
-        if resp["lastStatus"] == "STOPPED":
-            if self.grab_logs:
-                check_arn = resp["taskDefinitionArn"]
-                taskDefinition = self.ecs_client.describe_task_definition(taskDefinition=check_arn)[
-                    "taskDefinition"
-                ]
-                logconfig = taskDefinition["containerDefinitions"][0]["logConfiguration"]["options"]
-                streamobj = self.log_client.describe_log_streams(
-                    logGroupName=logconfig["awslogs-group"],
-                    logStreamNamePrefix=logconfig["awslogs-stream-prefix"],
-                )
-                logs = self.log_client.get_log_events(
-                    logGroupName=logconfig["awslogs-group"],
-                    logStreamName=streamobj["logStreams"][-1]["logStreamName"],
-                )
-                self.ecs_logs[offset] = logs["events"]
-                self.logs_messages[offset] = [log["message"] for log in logs["events"]]
-            self.tasks_done[offset] = True
-            return True
-        return False
+            retries -= 1
+            time.sleep(self.polling_interval)
 
-    def spin_til_done(self, offset=0, time_delay=5):
-        """
-        spinpolls the ECS servers to test if the task is done
+        if retries <= 0:
+            raise ECSTimeout()
 
-        Args:
-            offset (int): the offset of the task within the task array
-            time_delay (int): how long to wait between polls, seconds
-        """
-        while not self.check_if_done(offset=offset):
-            time.sleep(time_delay)
+        stop_codes = [task.get("stopCode") for task in tasks]
+        errors = [stop_code for stop_code in stop_codes if stop_code != "EssentialContainerExited"]
```

Comment:
This doesn't account for when tasks "successfully" stop but one or more of their containers return a non-zero exit code. As this code matures, I might change how this works to consider those to be failures as well.

Reply:
sounds reasonable, could we add a comment in the code for future tracking?